### ISSUE TITLE
`CREATE TABLE` should return empty DataFrame

### ISSUE THREAD

**Issue by andygrove** [apache/arrow-datafusion#3265]
**Describe the bug**

`CREATE TABLE` should just create the table and not run a query selecting the contents of the table.

```
❯ create table test as select 1;
+----------+
| Int64(1) |
+----------+
| 1        |
+----------+
```

This issue was already fixed for `CREATE VIEW` in https://github.com/apache/arrow-datafusion/issues/3236

**To Reproduce**
:point_up: 

**Expected behavior**
:point_up: 

**Additional context**
None

**Comment by kmitchener** [comment]
I'll do this one after #3251 is merged into master. that has a small refactoring of returning empty dataframes that this should use.
